js/genkit/src/genkit.ts

```diff
@@ -279,18 +279,12 @@ export class Genkit {
     name: string,
     options?: { variant?: string }
   ): Promise<ExecutablePrompt<z.infer<I>, O, CustomOptions>> {
-    // check if functional prompt exists.
-    let action = (await this.registry.lookupAction(
-      `/prompt/${name}`
-    )) as PromptAction<I>;
-    // if not, lookup with dotprompt.
-    if (!action) {
-      action = (await prompt(this.registry, name, options))
-        .promptAction as PromptAction<I>;
-    }
+    const dotprompt = await prompt(this.registry, name, options);
```

Comment:
The problem is this function can load dotprompt prompts from the registry ONLY. It will fail loading functional prompts.